c++/src/kj/mutex-test.c++

```diff
@@ -855,5 +859,40 @@ KJ_TEST("get location of shared mutex") {
 
 #endif
 
+#ifdef KJ_CONTENTION_WARNING_THRESHOLD
+KJ_TEST("make sure contended mutex warns") {
+  _::Mutex mutex;
+  LockSourceLocation exclusiveLockLocation;
+  mutex.lock(_::Mutex::EXCLUSIVE, nullptr, exclusiveLockLocation);
+
+  bool seenContendedLockLog = false;
+
+  auto threads = kj::heapArrayBuilder<kj::Own<kj::Thread>>(KJ_CONTENTION_WARNING_THRESHOLD);
+  for (auto i: kj::zeroTo(KJ_CONTENTION_WARNING_THRESHOLD)) {
+    (void)i;
+    threads.add(kj::heap<kj::Thread>([&mutex, &seenContendedLockLog]() {
+      _::LogExpectation expectation(LogSeverity::WARNING, "Acquired contended lock");
```

Comment:
`_::LogExpectation` is meant to implement `KJ_EXPECT_LOG`. It's kind of weird to go extend `KJ_EXPECT_LOG`'s privaet implementation to serve another purpose and then directly call into it from somewhere else... Maybe better to implement a custom `ExceptionCallback` for this test?

Reply:
Yeah, I was just being lazy.